### PR TITLE
[Windows] Teach Build-Testing how to find Foundation

### PR DESCRIPTION
**Pull request by rintaro**
Pass `Foundation_DIR` to the CMake configuration.

### CODE REVIEW

**Comment by rintaro** [comment]
https://github.com/swiftlang/swift-testing/pull/693
@swift-ci Please smoke test

**Comment by rintaro** [comment]
https://github.com/swiftlang/swift-testing/pull/693
@swift-ci Please build Toolchain

**Comment by rintaro** [comment]
https://github.com/swiftlang/swift-testing/pull/693
@swift-ci Please smoke test

**Comment by rintaro** [comment]
https://github.com/swiftlang/swift-testing/pull/693
@swift-ci Please smoke test

**Comment by rintaro** [comment]
https://github.com/swiftlang/swift-testing/pull/693
@swift-ci Please smoke test

**Comment by rintaro** [comment]
https://github.com/swiftlang/swift-testing/pull/693
@swift-ci Please test WebAssembly

**Comment by rintaro** [comment]
https://github.com/swiftlang/swift-testing/pull/693
@swift-ci Please test WebAssembly

**Comment by rintaro** [comment]
https://github.com/swiftlang/swift-testing/pull/693
@swift-ci Please smoke test

File: utils/build.ps1 [hunk 1]
```diff
@@ -1934,6 +1936,8 @@ function Build-Testing([Platform]$Platform, $Arch, [switch]$Test = $false) {
       -Defines (@{
         BUILD_SHARED_LIBS = "YES";
         CMAKE_BUILD_WITH_INSTALL_RPATH = "YES";
+        dispatch_DIR = "$DispatchBinaryCache\cmake\modules";
+        Foundation_DIR = "$FoundationBinaryCache\cmake\modules";
         SwiftSyntax_DIR = (Get-HostProjectCMakeModules Compilers);
```
Comment (compnerd) [inline utils/build.ps1:1941]: This seems suspect! But this is outside of your change I suppose.

**Review by compnerd — APPROVED** [review]

**Review by bnbarham — APPROVED** [review]

**Pull request merged**